documentation/modules/adding-source-provider.adoc

```diff
@@ -21,35 +21,161 @@ You can add {a-rhv} source provider by using the {ocp} web console.
+.. To validate a custom CA certificate, leave the *Skip certificate validation* check box cleared and _either_ drag the CA certificate to the text box _or_ browse for it and click *Select*.
+endif::[]
+ifdef::ova[]
+. Click *Open Virtual Appliance (OVA)*.
+. Specify the following fields:
+
+* *Provider Resource Name*: Name to display in the list of providers
+* *URL*: URL of the NFS file share that serves the OVA
+endif::[]
+ifdef::ostack[]
+. Click *OpenStack*.
+. Specify the following fields:
+
+* *Provider Resource Name*: Name to display in the list of providers.
+* *URL*: {osp} Identity (Keystone) endpoint, for example, `http://controller:5000/v3`.
+* *Authentication type*: Choose one of the following methods of authentication and supply the information related to your choice. For example, if you choose *Application credential ID* as the authentication type, the *Application credential ID* and the *Application Secret* fields become active, and you need to supply the ID and Secret.
+
+** *Application Credential ID*
+*** *Application Credential ID*: {osp} application credential ID
+*** *Application Credential Secret*: {osp} ahttps://github.com/kubev2v/forklift-documentation/pull/402pplication credential `Secret`
+
+** *Application Credential Name*
+*** *Application Credential Name*: {osp} application credential name
+*** *Application Credential Secret*: : {osp} application credential `Secret`
+*** *Username*: {osp} username
+*** *Domain*: {osp} domain
+
+** *Token with User ID*
+*** *Token*: {osp} token
+*** *User ID*: {osp} user ID
+*** *Project ID*:  {osp} project ID
+
+** *Token with User Name*
+*** *Token*: {osp} token
+*** *Username*: {osp} username
+*** *Project*: {osp} project
+*** *Domain Name*: {osp} domain name
+
+** *Password*
+*** *Username*: {osp} username
+*** *Password*: {osp} password
+*** *Project*: {osp} project
+*** *Domain*: {osp} domain
+
+. Choose one of the following options for validating a CA certificate:
+
+** *Skip validation* : Migrate without validating a CA certificate.
+** *Use the system CA certificate*: Migrate after validating the system CA certificate.
+** *Use a custom CA certificate*: Migrate after validating a custom CA certificate.
+
+.. To skip validation, select the *Skip certificate validation* check box.
+.. To validate the system CA certificate, leave the *Skip certificate validation* check box cleared and leave the *CA certificate* text box empty.
+.. To validate a custom CA certificate, leave the *Skip certificate validation* check box cleared and _either_ drag the CA certificate to the text box _or_ browse for it and click *Select*.
+endif::[]
+ifdef::cnv[]
+. Click *{virt}*.
+. Specify the following fields:
+
+* *Provider Resource Name*: Name to display in the list of providers
+* *URL*: Endpoint
+* *Service Account Token*: User or server account bearer token
+endif::[]
+ifdef::cnv2[]
+. Click *{virt}*.
```

Comment:
why do we have this twice?

Reply:
cnv is for OpenShift Virtualization as a source provider
cnv2 is for OpenShift Virtualization as a destination provider
Both use the same module with conditional text.